lib/cli.ex

```diff
@@ -22,30 +22,33 @@ defmodule Onigumo.CLI do
 
             case switches do
               [] -> module.main(working_dir)
-              _ -> usage_message()
+              _ -> usage_message("unknown options")
             end
 
           :error ->
-            usage_message()
+            usage_message("unknown positional argument")
         end
 
       {_, _, [_ | _]} ->
         usage_message()
 
       {_, argv, _} when length(argv) != 1 ->
-        usage_message()
+        usage_message("exact one positional argument must be used")
     end
   end
 
-  defp usage_message() do
+  defp usage_message(reason \\ "") do
     IO.write("""
-    onigumo: invalid usage
+    onigumo: invalid usage#{reason_message(String.trim(reason))}
     Usage: onigumo [OPTION]... [COMPONENT]
 
     Try `onigumo --help' for more options.
     """)
   end
 
+  defp reason_message(reason) when byte_size(reason) == 0, do: reason
+  defp reason_message(reason), do: ", #{reason}"
+
```

Comment:
This may be too complicated. If we make sure there is always a reason, we don’t need this function.

```suggestion
```